wrappers/python/tests/layer/test_gpt2_attention.py

```diff
@@ -0,0 +1,201 @@
+   
+    nntile_layer, _ = nntile.layer.GPT2Attention.from_torch(
+            torch_layer, X, X, X, nntile_config, 0
+    )
+
+    y_grad_random = rng.standard_normal(nntile_layer.y.grad.shape)
+    y_grad_nntile = np.array(y_grad_random, dtype=np.float32, order="F")
+    nntile_layer.y.grad.from_array(y_grad_nntile)
+    y_grad_torch = torch.Tensor(y_grad_nntile.T)
+    return torch_layer, nntile_layer, x_torch, y_grad_torch
+
+
+@pytest.mark.parametrize('params', [
+    pytest.param(single_tile, id='single_tile'),
+    pytest.param(multiple_tiles, id='multiple_tiles'),
+])
+@pytest.mark.parametrize('dtype', [
+    'fp32',
+    pytest.param('fp32_fast_tf32', marks=nocuda),
+    pytest.param('bf16', marks=nocuda),
+])
+
+class TestGPT2Attention:
+
+    def test_torch_coercion(self, starpu_simple, torch_rng, dtype: str,
+                            params: GPT2AttentionTestParams):
+        torch_layer, nntile_layer, *_ = generate_inputs(dtype, params)
+        torch_layer_other = nntile_layer.to_torch()
+        nntile_layer.unregister()
+        nntile_layer.x_q.unregister()
+        nntile_layer.x_k.unregister()
+        nntile_layer.x_v.unregister()
+        nntile_layer.y.unregister()
+
+        rtol = dtype2tol[dtype]['rtol']
+        for (n1, p1), (n2, p2) in zip(torch_layer.named_parameters(),
+                torch_layer_other.named_parameters()):
+            assert n1 == n2
+            assert torch.norm(p1 - p2) <= rtol * torch.norm(p1)
+
+    def test_forward(self, starpu_simple, torch_rng, dtype: str,
+                     params: GPT2AttentionTestParams):
+        torch_layer, nntile_layer, x, _ = generate_inputs(dtype, params)
+        y, _ = torch_layer(x)
+        nntile_layer.forward_async()
+        y_nntile = torch.Tensor(to_numpy(nntile_layer.y.value).T)
+        nntile_layer.unregister()
+        nntile_layer.x_q.unregister()
+        nntile_layer.x_k.unregister()
+        nntile_layer.x_v.unregister()
+        nntile_layer.y.unregister()
+        print(torch.norm(y_nntile), torch.norm(y))
+        rtol = dtype2tol[dtype]['rtol']
+        assert torch.norm(y - y_nntile) <= rtol * torch.norm(y)
+
+    def test_forward_backward(self, starpu_simple, torch_rng, dtype: str,
+                              params: GPT2AttentionTestParams):
+        torch_layer, nntile_layer, x, y_grad = generate_inputs(dtype, params)
+        y, _ = torch_layer(x)
+        nntile_layer.forward_async()
+        y_nntile = torch.Tensor(to_numpy(nntile_layer.y.value).T)
+        res = (y * y_grad).sum()
+        res.backward()
+        nntile_layer.backward_async()
+        torch_layer_other = nntile_layer.to_torch_with_grads()
+        nntile_layer.unregister()
+        nntile_layer.x_q.unregister()
+        nntile_layer.x_k.unregister()
+        nntile_layer.x_v.unregister()
+        nntile_layer.y.unregister()
+
+        rtol = dtype2tol[dtype]['rtol']
+        assert torch.norm(y - y_nntile) <= rtol * torch.norm(y)
```

Comment:
Instead of checking output, check gradient of inputs. This checking of output in `test_backward` was my mistake. I forgot to switch it to checking of gradient of inputs.